feature_engine/creation/distance_features.py

```diff
@@ -0,0 +1,241 @@
+        output_column_names=['distance_between_a_and_b'],
+        drop_original=False,
+    )
+    >>> cf.fit(X)
+    >>> cf.transform(X)
+       a_latitude   a_longitude   b_latitude   b_longitude     distance_between_a_and_b
+    0       0.           0.           0.           0.                   0.
+    1       0.           0.          12.34       123.45             13630.28
+    2      46.94         7.43        59.91        10.75              1457.49
+    """
+
+    _EARTH_RADIUS: float = 6371.0  # radius of Earth in kms
+
+    def __init__(
+        self,
+        coordinate_columns: List[List[Union[str, int]]],
+        output_column_names: Optional[List[Union[str, None]]] = None,
+        drop_original: bool = False,
+    ) -> None:
+
+        # the coordinate_columns variable is rewritten in this way to speed up
+        # computation later, i.e., to use vectorization
+        (self.a_latitudes, self.a_longitudes, self.b_latitudes, self.b_longitudes) = (
+            self._check_coordinate_columns(columns=coordinate_columns)
+        )
+        self.output_column_name = self._check_output_columns_names(
+            column_name=output_column_names,
+            coordinate_columns=coordinate_columns,
+        )
+
+        self.drop_original = self._check_drop_original(parameter=drop_original)
+
+        self.variables = None
+
+    def _check_drop_original(self, parameter: bool) -> bool:
+        if isinstance(parameter, bool) is False:
+            raise ValueError(
+                "Expected boolean value for parameter `drop_original`, "
+                f"but got {parameter} with type {type(parameter)}"
+            )
+        return parameter
+
+    def _check_coordinate_columns(
+        self, columns: List[List[Union[str, int]]]
+    ) -> Tuple[List[Union[str, int]], ...]:
+        if not columns:
+            raise ValueError("Empty list for `coordinate_columns` not allowed!")
+        for idx, coordinate_column in enumerate(columns):
+            if len(coordinate_column) != 4:
+                raise ValueError(
+                    f"Needed 4 values to compute a distance, "
+                    f"but got {len(coordinate_column)} columns \n"
+                    f"at the index {idx} of the list coordinate columns."
+                )
+        return (
+            [coordinate[0] for coordinate in columns],
+            [coordinate[1] for coordinate in columns],
+            [coordinate[2] for coordinate in columns],
+            [coordinate[3] for coordinate in columns],
+        )
+
+    def _check_output_columns_names(
+        self,
+        column_name: Optional[List[Union[str, None]]],
+        coordinate_columns: List[List[Union[str, int]]],
+    ) -> Optional[List[Union[str, None]]]:
+        if column_name is None:
+            return [f"distance_{c[0]}_{c[1]}_{c[2]}_{c[3]}" for c in coordinate_columns]
+        if len(column_name) != len(coordinate_columns):
```

Comment:
I don't understand this check, we are returning only 1 distance variable, so we don't need to check how many names, or we need to check that it is only 1 name